nums/core/linalg.py

```diff
@@ -256,6 +256,160 @@ def inv(app: ArrayApplication, X: BlockArray):
     return result
 
 
+def inv_uppertri(app: ArrayApplication, X: BlockArray):
+    # Inversion of an Upper Triangular Matrix
+    # Use the method described in https://www.cs.utexas.edu/users/flame/pubs/siam_spd.pdf
+    assert X.shape[0] == X.shape[1], "This function only accepts square matrices"
```

Comment:
```suggestion
    assert len(X.shape) == 2
    assert X.shape[0] == X.shape[1], "This function only accepts square matrices"
```

Reply:
Apart from these, are there any assumptions you have made for the supported matrix types. Just want to be sure all the appropriate assertions are in place for what we support currently

Reply:
Yes, besides the square matrix assumption, we are also making the assumption that the matrix is upper-triangular. I could implement validation for that, but would that be too much overhead? especially if `X` is super large?

Reply:
hmm. what do you think @elibol